### PR TITLE
Add service_types support for neutron subnet

### LINKED ISSUE #2478: Support for service types on neutron subnets
service types can be set on create and update of a neutron subnet but it is not supported yet.
[Docs](https://docs.openstack.org/api-ref/network/v2/index.html?expanded=create-subnet-detail,update-subnet-detail#subnets)

### PR DESCRIPTION
**Pull request by nikParasyr**
Fixes #2478 

[Docs](https://docs.openstack.org/api-ref/network/v2/index.html?expanded=create-subnet-detail,update-subnet-detail#subnets)

### CODE REVIEW

**Comment by nikParasyr** [comment]
@mandre this is rdy from my side. i cant fix the PR label job. There are more PRs with this job failing as well

File: openstack/networking/v2/subnets/doc.go [hunk 1]
```diff
@@ -98,11 +99,13 @@ Example to Update a Subnet
 
 	subnetID := "db77d064-e34f-4d06-b060-f21e28a61c23"
 	dnsNameservers := []string{"8.8.8.8"}
+	serviceTypes := []string{"network:floatingip", "network:routed"}
 	name := "new_name"
 
 	updateOpts := subnets.UpdateOpts{
 		Name:           &name,
 		DNSNameservers: &dnsNameservers,
+		ServiceTypes:   &ServiceTypes,
```
Comment (mandre) [inline openstack/networking/v2/subnets/doc.go:108]:
```suggestion
		ServiceTypes:   &serviceTypes,
```

**Review by mandre — COMMENTED** [review]
LGTM, just the small typo in the docs.

File: openstack/networking/v2/subnets/doc.go (hunk 1 above)
Comment (nikParasyr) [reply openstack/networking/v2/subnets/doc.go]: ty. fixed

**Review by mandre — APPROVED** [review]
Thanks!

**Pull request merged**